modules/xfem/src/base/XFEMSquareCut.C

```diff
@@ -0,0 +1,67 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "XFEMSquareCut.h"
+
+#include "MooseError.h"
+#include "XFEMFuncs.h"
+#include "EFAFuncs.h"
+
+XFEMSquareCut::XFEMSquareCut(std::vector<Real> square_nodes):
+  XFEMGeometricCut3D(0.0, 0.0),
+  _vertices(4, Point(0.0,0.0,0.0))
+{
+  _vertices[0] = Point(square_nodes[0], square_nodes[1], square_nodes[2]);
+  _vertices[1] = Point(square_nodes[3], square_nodes[4], square_nodes[5]);
+  _vertices[2] = Point(square_nodes[6], square_nodes[7], square_nodes[8]);
+  _vertices[3] = Point(square_nodes[9], square_nodes[10], square_nodes[11]);
+
+  for (unsigned int i = 0; i < 4; ++i)
+    _center += _vertices[i];
+  _center *= 0.25;
+
+  for (unsigned int i = 0; i < 4; ++i)
+  {
+    unsigned int iplus1(i < 3 ? i+1 : 0);
```

Comment:
Sure, but then we couldn't use the ternary operator here, so this would take multiple lines of code. We do this all over the place. This seems better to me, no?